src/main/java/frc/robot/utils/CRTGearRatioUtil.java

```diff
@@ -0,0 +1,154 @@
+   * @param secondaryGearRotationRatioTerm for every mainGearRotationRatioTerm
+   *                                       rotations of the main gear,
+   *                                       the secondary gear rotates
+   *                                       secondaryGearRotationRatioTerm times
+   * @param maxRotations                   the maximum number of rotations that
+   *                                       the main gear will make that involve
+   *                                       unique positions of the object
+   *                                       attached to it
+   * @param mainGearCircumference          circumference of the main gear
+   * 
+   * @return Returns optional of the util if the gear ratio works with the number
+   *         of max rotations, otherwise returns an empty optional
+   */
+  public static Optional<CRTGearRatioUtil> init(int mainGearRotationRatioTerm, int secondaryGearRotationRatioTerm,
+      int maxRotations, double mainGearCircumference) {
+    CRTGearRatioUtil u = new CRTGearRatioUtil(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm, maxRotations,
+        mainGearCircumference);
+    if (u.validateGearRatioWithMaxRotations()) { //check if the gear ratio works with the required amount of rotations by the main gear
+      return Optional.of(u);
+    }
+    return Optional.empty();
+  }
+
+  private CRTGearRatioUtil(int mainGearRotationRatioTerm, int secondaryGearRotationRatioTerm, int maxRotations,
+      double mainGearCircumference) {
+    this.mainGearRotationRatioTerm = (double) mainGearRotationRatioTerm;
+    this.secondaryGearRotationRatioTerm = (double) secondaryGearRotationRatioTerm;
+    this.mainGearRotationRatioTermSimplified = (double) (mainGearRotationRatioTerm
+        / gcd(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm));
+    this.secondaryGearRotationRatioTermSimplfied = (double) (secondaryGearRotationRatioTerm
+        / gcd(mainGearRotationRatioTerm, secondaryGearRotationRatioTerm));
+    this.maxRotations = maxRotations;
+    this.mainGearCircumference = mainGearCircumference;
+  }
+
+  //"These two functions speak for themselves" - Hans Niemann commenting his code
+  public static double gcd(double a, double b) {
+    if (b == 0.0)
+      return a;
+    return gcd(b, a % b);
+  }
+
+  public static double mod(double a, double b) {
+    double r = a % b;
+    if (r < 0) {
+      r += b;
+    }
+    return r;
+  }
+
+  /**
+   * 
+   * @return Returns a multiplier used by CRT. You shouldn't need to use this function yourself, and
+   *         I don't fully understand it (Victor Chen did the math behind it).
+   */
+  public int kFactor() {
+    double sum = 0;
+    double a = secondaryGearRotationRatioTermSimplfied;
+    double b = mainGearRotationRatioTermSimplified;
+    for (int i = 1; i < b; i++) {
+      sum += ((a * (b - i) - 1.0) / b)
+          * (1 + Math.floor(((a * (b - i) - 1.0) / b)) - Math.ceil(((a * (b - i) - 1.0) / b)));
```

Comment:
`(a * (b - i) - 1.0) / b)` this is used three times, maybe split it out into a variable. Also what is even happening here wow

Reply:
I asked Victor about this and he said he would try to explain it more to me at some point this week. I'll put that into a variable.